lib/ClangImporter/ImporterImpl.h

```diff
@@ -248,14 +248,20 @@ enum class FactoryAsInitKind {
 
 namespace importer {
 struct PlatformAvailability {
-  /// A predicate that indicates if the given platform should be
-  /// considered for availability.
-  std::function<bool(StringRef PlatformName)> filter;
-
-  /// A predicate that indicates if the given platform version should
-  /// should be included in the cutoff of deprecated APIs marked unavailable.
-  std::function<bool(unsigned major, llvm::Optional<unsigned> minor)>
-      deprecatedAsUnavailableFilter;
+private:
+  PlatformKind platformKind;
+
+public:
+  /// Returns true when the given platform should be considered for
+  /// availabilityon imported declarations.
+  bool isPlatformRelevant(StringRef platform) const;
+
+  /// Returns true when the given declaration with the given deprecation
+  /// should be inlucded in the cutoff of imported deprecated APIs marked
+  /// unavailable.
+  bool treatDeprecatedAsUnavailable(const clang::Decl *clangDecl,
+                                    unsigned major,
```

Comment:
Is there a reason to pass the major and minor numbers as separate parameters instead of a single `llvm::VersionTuple`? It looks like all the call sites you touched here extract the values from a `VersionTuple`, so that might simplify things a bit.

Reply:
Good idea! Updated to use use version tuple.